### ISSUE TITLE
Error Did not find the file passed to "--dart-define-from-file" on extraArgs

### ISSUE THREAD

**Issue by diegolmauro** [hey24sheep/azure-flutter-tasks#125]
Hi.

I'm fallowing the instructions of the answered of this issue: [https://github.com/hey24sheep/azure-flutter-tasks/issues/84](https://github.com/hey24sheep/azure-flutter-tasks/issues/84), and I'm trying to use `--dart-define-from-file` by passing it to `extraArgs`, but I always get the error `Did not find the file passed to "--dart-define-from-file"`. I've tried several patterns, such as "/configs/dev.json", "$(Build.SourcesDirectory)/configs/dev.json" and "configs/dev.json". Can you please help me with this?

**Comment by hey24sheep** [comment]
Hi, I will check and see what can be done. I am busy with work and life, give me some time and I will push a fix ASAP

**Comment by hey24sheep** [comment]
Hi @diegolmauro, I have updated the version from **0.3.18** to **0.4.0** to denote major changes (this fixes #124). 

If previous version of this extension is needed then you download it from [releases](https://github.com/hey24sheep/azure-flutter-tasks/releases/tag/0.3.18_node10) 

For a week I will leave the extension as is on the latest update to validate everything is working fine with the updated node & packages. Next week, I will try my best to push a fix for this (I do not have access to Flutter/Azure project anymore so most of my fixes are blind, so bear with me here)

Thanks

**Comment by hey24sheep** [comment]
Can you try one thing, - check this https://stackoverflow.com/questions/77445120/flutter-dart-define-from-file-is-not-working

and try `extraArgs` as 
- `--dart-define-from-file="lib/config.json"`
- `--dart-define-from-file="./lib/config.json"`
if you have other arguments, use the "extraArgsSeparator"

**Comment by diegolmauro** [comment]
Hi @hey24sheep, thank you for the quick response.

Here’s the structure of my project’s path:

```
main folder
   |_ configs
       |_ dev.json
   |_ lib
       |_ main.dart
```

I've tried all the suggested patterns and using both `../configs/dev.json` and `~/configs/dev.json`  patterns, but none of them work when running from the FlutterBuild task. However, when I run the flutter build command directly from a bash script, everything works fine with the `configs/dev.json` pattern.

If I move the `configs` folder into the `lib` folder, it works. 

Thanks.
Best regards.

**Comment by hey24sheep** [comment]
Can you try move config to lib folder and try with the build task, does that work? 

Actually, it should work with extraArgs. I'm wondering why it isn't working

**Comment by diegolmauro** [comment]
Yes, moving the `configs` folder to the `lib` directory works perfectly with the build task. However, running the `flutter build` command directly works fine even with the original folder structure.

```
main folder
   |_ configs
       |_ dev.json
   |_ lib
       |_ main.dart
```

**Comment by hey24sheep** [comment]
Okay, I'm sorry but I'm a little confused now. Is it working now with extraArgs? With the current build task? Because I don't want to add one more configuration option if it's working.

**Comment by diegolmauro** [comment]
The build task only works when I move the `configs` folder to the `lib` folder. If I keep my original folder structure, it doesn't work.

What I'm trying to explain is that when using the `flutter build` command directly in a script (instead of the build task), it works in both cases, even when the `configs` folder is kept in the root directory.

**Comment by hey24sheep** [comment]
Ohh, I got it now. So, then if I add one more option it could be a shot in the dark. It might work or won't. I could try and add during the weekend but I'm pretty sure it won't work because that argument will be passed the same way as extraArgs is processing it right now.
